src/document/DocumentManager.ts

```diff
@@ -67,7 +68,13 @@ export class DocumentManager {
   }
   async onFileUpload(file: File | null) {
     await this.parseFile(file)
-      .then((content) => this.model.fromSavedDocument(JSON.parse(content)))
+      .then((content) => {
+        if (VERSIONS["v0.1"].validate(JSON.parse(content))) {
```

Comment:
I forget exactly how the plumbing goes here but validation (using the correct document version) ought to happen before the automatic document upgrade chain